_delphi_utils_python/delphi_utils/logger.py

```diff
@@ -33,21 +32,19 @@ def threading_exception_handler(args):
         if args.exc_type == SystemExit and args.exc_value.code == 0:
             # `sys.exit(0)` is considered "successful termination":
             #   https://docs.python.org/3/library/sys.html#sys.exit
-            logger.debug("normal thread exit", thread=args.thread,
-                        stack="".join(
-                            format_exception(
-                                args.exc_type, args.exc_value, args.exc_traceback)))
+            logger.debug(
+                "normal thread exit",
+                thread=args.thread,
+                stack="".join(format_exception(args.exc_type, args.exc_value, args.exc_traceback)),
+            )
         else:
-            exception_handler(f"thread: {args.thread}",
-                              args.exc_type, args.exc_value, args.exc_traceback)
+            exception_handler(f"thread: {args.thread}", args.exc_type, args.exc_value, args.exc_traceback)
 
     sys.excepthook = sys_exception_handler
     threading.excepthook = threading_exception_handler
 
 
-def get_structured_logger(name=__name__,
-                          filename=None,
-                          log_exceptions=True):
+def get_structured_logger(name=__name__, filename=None, log_exceptions=True, is_real=True):
```

Comment:
```suggestion
def get_structured_logger(name=__name__, filename=None, log_exceptions=True, *, test_mode=False):
```
and then below:
```py
cache_logger_on_first_use=not(test_mode),
```

This name (and inverted truth value) makes the purpose of the argument clearer, and the `*` in the arguments list ensures its only ever triggered explicitly.